P5-243 og anden: Rettelser til Analysis og opsætning af Solution chapter. (#157)
aau-hcl-p5/main: report/5.Solution/pythoncontroller.tex

\section{Python} \label{pythonstructure}

> The python part of the project, is separated into multiple modules.

Reviewer: ```suggestion
The Python part of the project is separated into multiple modules.
```

> The intention of the modules is to allow for a system, that can be easily modified, based on the specific context.

Reviewer: ```suggestion
The intention of a modular structure is to allow for a system that can be easily modified, based on the specific context.
```

optional arguments:
-h, --help            
	show this help message and exit
-a [name], --algorithm [name] 
	Choose which algorithm to run [GOTURN, YOLO, ZONE_AVG, OBJ_FILL]
	default='obj_fill'
-n [no_usb], --no-usb [no_usb]
	Whether to disable USB connection with the NXT.
	default=False
\end{lstlisting}
\todo{make sure these are still up to date when finishing the project}

The main objective of \texttt{main.py} is to determine which arguments to pass onto the \texttt{FlatController}.

\subsection{FlatController}\label{flatcontrollerimplementation}
The \texttt{FlatController} class manages the different modules of the python part of the project and ties the python part of the solution together
The signature of the \texttt{FlatController} is shown in Snippet~\ref{lst:FlatControllerInit}.

\begin{lstlisting}[language=Python,label={lst:FlatControllerInit},caption={Initialization method of the \texttt{FlatController} class}]
	def __init__(self,
		algorithm: Callable[[np.ndarray], Optional[Vector]],
		output_device: OutputDevice,
		input_device: VideoController,
		calibration_algorithm: Union[Callable[[OutputDevice], None], None] = None,
		debug=True
		) -> None:
\end{lstlisting}

> The order of the parameters are not an indication of their relevance, but is rather based on the requirement of python itself, as python requires parameters with default parameters to be first in the signature of the methods.

Reviewer: ```suggestion
The order of the parameters are not an indication of their relevance, but is rather based on the requirement of python itself, as Python requires parameters with default parameters to be first in the signature of the methods.
```
